lighthouse-core/runner.js

```diff
@@ -153,6 +153,52 @@ class Runner {
         .readdirSync(path.join(__dirname, './gather/gatherers'))
         .filter(f => /\.js$/.test(f));
   }
+
+  /**
+   * Resolves the location of the specified plugin and returns an absolute
+   * string path to the file. Used for loading custom audits and gatherers.
+   * Throws an error if no plugin is found.
+   * @param {string} plugin
+   * @param {string=} configDir The absolute path to the directory of the config file, if there is one.
+   * @param {string=} category Optional plugin category (e.g. 'audit') for better error messages.
+   * @return {string}
+   * @throws {Error}
+   */
+  static resolvePlugin(plugin, configDir, category) {
```

Comment:
can't use default parameters in LH yet since we still support node v4 and v5. Looking forward to #449 :)